src/System.Management.Automation/engine/Utils.cs

```diff
@@ -2278,4 +2278,34 @@ internal ReadOnlyBag(HashSet<T> hashset)
         /// </summary>
         internal static readonly ReadOnlyBag<T> Empty = new ReadOnlyBag<T>(new HashSet<T>(capacity: 0));
     }
+
+    /// <summary>
+    /// Helper class for simple argument validations.
+    /// </summary>
+    internal static class Requires
+    {
+        internal static void NotNull(object value, string paramName)
```

Comment:
```suggestion
        internal static void NotNull([NotNull] object? value, string paramName)
```

Reply:
`[NotNull]` means the parameter must not be null, but it surely can be null. Also, nullable is not enforced in this file, so I would rather not use the nullable syntax here.

Reply:
> [NotNull] means the parameter must not be null, but it surely can be null.

It looks a trick but it works as I suggested - if the method returns the value is not null. 

> Also, nullable is not enforced in this file, so I would rather not use the nullable syntax here.

The class looks very useful and I'd expect we will use it often. So I suggest to enable nullable annotation locally for the class.

Reply:
It looks wrong so it will affect readability.
Also, I don't think it's that useful to use nullable in this class. If nullable is enabled in a file and a parameter is supposed to be not null, then just declare that parameter as no-nullable, and then use `Requires.NotNull` to validate it, for example:
```c#
        public static SubsystemInfo GetSubsystemInfo(Type subsystemType)
        {
            Requires.NotNull(subsystemType, nameof(subsystemType));
            ....
        }
```